cabal-install/src/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -175,6 +177,20 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
             ] ++
             bin_file (compSolverName comp)
      where
+      -- | Only add build-info file location if the Setup.hs CLI
+      -- is recent enough to be able to generate build info files.
+      -- Otherwise, do not add the expected file location.
+      --
+      -- Consumers of `plan.json` can use the absence of this file location
```

Comment:
These docs need to be updated to reflect the fact that we null the field now instead of leaving it out.